MoreLinq.Test/OrderedMergeTest.cs

```diff
@@ -0,0 +1,398 @@
+        }
+
+        [Test]
+        public void ShouldDisposeEnumerators()
+        {
+            using (var first = TestingSequence.Of(new int[] { }))
+            using (var second = TestingSequence.Of(new int[] { }))
+                MoreEnumerable.OrderedMerge(first, second, id => id, id => id, id => id, id => id, (f, _) => f, null).ToArray();
+        }
+
+        [Test]
+        public void IfThereAreNoMoreElementsToReturnFromTheFirstCollectionThenReturnTheRemainingSecondCollection()
+        {
+            var first = new int[] { };
+            var second = new[] { 1, 2, 3 };
+
+            int Id(int id) => id;
+            int First(int f, int _) => f;
+
+            var merged = MoreEnumerable.OrderedMerge(
+                first: first,
+                second: second,
+                firstKeySelector: Id,
+                secondKeySelector: Id,
+                firstSelector: Id,
+                secondSelector: Id,
+                bothSelector: First,
+                comparer: null);
+
+            Assert.That(merged, Is.EqualTo(second));
+        }
+
+        [Test]
+        public void IfThereAreNoMoreElementsToReturnFromTheSecondCollectionThenReturnTheRemainingFirstCollection()
+        {
+            var first = new[] { 1, 2, 3 };
+            var second = new int[] { };
+
+            int Id(int id) => id;
+            int First(int f, int _) => f;
+
+            var merged = MoreEnumerable.OrderedMerge(
+                first: first,
+                second: second,
+                firstKeySelector: Id,
+                secondKeySelector: Id,
+                firstSelector: Id,
+                secondSelector: Id,
+                bothSelector: First,
+                comparer: null);
+
+            Assert.That(merged, Is.EqualTo(first));
+        }
+
+        [Test]
+        public void TwoSequencesWithNoCollisionsShouldMergeUsingTheDefaultComparer()
+        {
+            var first = new[] { 1, 4, 5 };
+            var second = new[] { 2, 3, 6 };
+
+            int Id(int id) => id;
+            int First(int f, int _) => f;
+
+            var merged = MoreEnumerable.OrderedMerge(
+                first: first,
+                second: second,
+                firstKeySelector: Id,
+                secondKeySelector: Id,
+                firstSelector: Id,
+                secondSelector: Id,
+                bothSelector: First,
+                comparer: null);
+
+            Assert.That(merged, Is.EqualTo(new[] { 1, 2, 3, 4, 5, 6 }));
+        }
+
+        [Test]
+        public void WhenThereIsACollisionShouldUseBothSelectorToChooseFromFirstOfSecondCollection()
+        {
+            var firstElement = new Version(3, 0);
+            var secondElement = new Version(3, 0);
+
+            var first = new[] { firstElement };
+            var second = new[] { secondElement };
+            Version First(Version f, Version _) => f;
```

Comment:
All occurrences of `First` are used once and therefore should be inlined to simply read any of the following:

- `(f, _) => f`
- `(fst, _) => fst`
- `(first, _) => first`

Same goes for `Second` on the next line.